packages/core/src/builtin-tasks/moleculec.ts

```diff
@@ -0,0 +1,282 @@
+import { execSync } from 'node:child_process'
+import fs from 'node:fs'
+import path from 'node:path'
+import { task } from '../config/config-env'
+import { paramTypes } from '../params'
+
+interface Args {
+  schemaFile: string
+  language: 'rust' | 'c' | 'ts'
+  output?: string
+}
+
+const extMap: Record<Args['language'], string> = {
+  rust: '.rs',
+  c: '.cpp',
+  ts: '.ts',
+}
+
+enum UInts {
+  Uint8 = 1,
+  Uint16 = 2,
+  Uint32 = 4,
+  Uint64 = 8,
+  Uint128 = 16,
+  Uint256 = 32,
+  Uint512 = 64,
+}
+
+const basicMoleculeType = 'byte' as const
+
+type SimpleMoleculeType = keyof typeof UInts
+
+const tsBasicMolecuelType: SimpleMoleculeType = 'Uint8'
+
+type MoleculeInterface =
+  | SimpleMoleculeType
+  | TableInterface
+  | StructInterface
+  | UinonInterface
+  | VecInterface
+  | ArrayInterface
+  | OptionInterface
+
+interface TableDeclare {
+  type: 'table'
+  fields: {
+    name: string
+    type: string
+  }[]
+}
+
+interface TableInterface {
+  type: 'table'
+  value: Record<string, MoleculeInterface>
+}
+
+interface StructDeclare {
+  type: 'struct'
+  fields: {
+    name: string
+    type: string
+  }[]
+}
+
+interface StructInterface {
+  type: 'struct'
+  value: Record<string, MoleculeInterface>
+}
+
+interface UinonDeclare {
+  type: 'union'
+  items: {
+    typ: string
+    id: number
+  }[]
+}
+
+interface UinonInterface {
+  type: 'union'
+  value: Record<string, MoleculeInterface>
+}
+
+interface VecDeclare {
+  type: 'fixvec' | 'dynvec'
+  item: string
+}
+
+interface VecInterface {
+  type: 'vec'
+  value: MoleculeInterface
+}
+
+interface ArrayDeclare {
+  type: 'array'
+  item: string
+  item_count: number
+}
+
+interface ArrayInterface {
+  type: 'array'
+  value: [MoleculeInterface, number]
+}
+
+interface OptionDeclare {
+  type: 'option'
+  item: string
+}
+
+interface OptionInterface {
+  type: 'option'
+  value: MoleculeInterface
+}
+
+type Declaration = (ArrayDeclare | StructDeclare | VecDeclare | TableDeclare | UinonDeclare | OptionDeclare) & {
+  name: string
+  imported_depth: number
+}
+
+interface Schema {
+  namespace: string
+  declarations: Declaration[]
+}
+
+function generateFixBytes(itemCount: number) {
+  return (
+    (UInts[itemCount] as SimpleMoleculeType) ??
```

Comment:
`SimpleMoleculeType` can be removed because `UInts` is typed

Reply:
typeof `UInts[itemCount]` is string, but not `'Uint8' | 'Uint16' ....`